src/tutorials/jbearing2.c

```diff
@@ -583,6 +587,5 @@ PetscErrorCode CallPermonAndCompareResults(Tao tao, void *ctx)
     test:
       suffix: 6
       nsize: 3
-      args: -mx 50 -my 50
+      args: -mx 30 -my 30
 TEST*/
-
```

Comment:
Please keep the empty line at the end of the code because some compilers (CRAY) show a warning when the file is not POSIX compliant.

Reply:
* I'm not aware I would do this explicitly, maybe GitHub or some other tool did that automatically.
* I don't think you're right here. The source file should end with `\n`. Which Vim does by default, I think. No need for blank line - that would actually be `\n\n`.
https://unix.stackexchange.com/a/18789
* PETSc does not have blank line at EOF everywhere: 
  - https://gitlab.com/petsc/petsc/-/blob/master/src/dm/impls/plex/tests/ex1.c
  - https://gitlab.com/petsc/petsc/-/blob/master/src/ksp/ksp/interface/itcreate.c
* Git actually deems it a whitespace error:
```
   $ git diff
diff --git a/src/tutorials/jbearing2.c b/src/tutorials/jbearing2.c
index 23244b3..e4adbac 100644
--- a/src/tutorials/jbearing2.c
+++ b/src/tutorials/jbearing2.c
@@ -589,3 +589,4 @@ PetscErrorCode CallPermonAndCompareResults(Tao tao, void *ctx)
       nsize: 3
       args: -mx 30 -my 30
 TEST*/
+
$ # + above is printed with red background
$ git diff --check
src/tutorials/jbearing2.c:592: new blank line at EOF.
```

Reply:
So I think the blank line before EOF should actually be considered erroneous rather than required :-)

Reply:
> `# + above is printed with red background`
> `new blank line at EOF`

On the newest git, I get this warning only when there is 3 or more successive \n. I.e., having `text\n\nEOF` is considered OK.

Anyway, I agree in this case there would be a double \n, which is unnecessary (but not erroneous). You can keep it as it is.

Both git diff and GitHub web interface highlight missing newlines sufficiently so that there is no need for double \n. (`\ No newline at end of file` in git diff and see ex1.c for GitHub example in https://github.com/permon/permonsvm/pull/1/commits/78ff1ad23b9b6db256da322c02bb2118f2f4119b).